### PR TITLE
Implement navigation in image viewer

### LINKED ISSUE #3522: Previous/next image buttons in the image viewer
Vue branch currently does not implement these, and I'm making this separate issue for it because I'm more than certain it won't be implemented in the `vue` branch.

It was initially implemented in TL in #1365

I want to know whether people use these buttons, and if they do, does it work as expected? In over 2 years of these existing I have not used them once.

Furthermore, the imageviewer in vue implements a lot more interactions with the image (zooming, moving around, etc), and adding arrow buttons might affect that behaviour.

### PR DESCRIPTION
**Pull request by richrd**
Fixes #3522

### CODE REVIEW

File: client/components/ImageViewer.vue [hunk 1]
```diff
@@ -78,8 +105,34 @@ export default {
 			}
 
 			this.$root.$off("resize", this.correctPosition);
+			this.channel = null;
+			this.previousImage = null;
+			this.nextImage = null;
 			this.link = null;
 		},
+		getNewImage(delta) {
```
Comment (xPaw) [inline client/components/ImageViewer.vue]: Better off assigning next/prev within this once, so it doesn't have to filter twice.

**Comment by xPaw** [comment]
cc @dgw for testing

**Comment by xPaw** [comment]
10-1325-MistyrosePuma

The arrow buttons don't overlay particularly well over the image. I believe the old image viewer had padding to account for them.

Perhaps they should be repositioned?

**Comment by richrd** [comment]
Yeah I think we want to avoid showing them on top of the image at least by default, unless the user zooms it. Alternatively we could only show them for a short while on hover and hide them after a timeout, similarly as the controls of video players usually do.

**Comment by xPaw** [comment]
We could also reposition them to the bottom for example.

**Comment by maxpoulin64** [comment]
Just checked with the old one for comparison, and the arrows were just always on top of the image in the same position: 

image

I think this is fine to accept as-is at least for feature parity with the older one, and it can be improved separately.

Code looks fine to me, but I'm not confident enough with the new Vue thing to give it a full review/approval.

**Comment by richrd** [comment]
If that's how it was in the old version, I agree, we can improve later if we feel like it.

File: client/components/ImageViewer.vue [hunk 2]
```diff
@@ -56,30 +76,62 @@ export default {
...
+
+			if (!oldLink) {
+				this.$root.$on("escapekey", this.closeViewer);
+				this.$root.$on("resize", this.correctPosition);
+				Mousetrap.bind("left", this.previous);
+				Mousetrap.bind("right", this.next);
+			}
 		},
 	},
-	mounted() {
```
Comment (xPaw) [inline client/components/ImageViewer.vue:68]: Failed rebase?

Comment (richrd) [reply client/components/ImageViewer.vue:68]: I think this is covered by the watch, or is it?

Comment (xPaw) [reply client/components/ImageViewer.vue:68]: Oh this PR includes a commit which is already on master. Can you rebase it cleanly?

Comment (xPaw) [reply client/components/ImageViewer.vue:68]: I didn't realize my commit wasn't on master, so this is fine.

**Pull request merged**